src/Package/Impl/Shell/VsAppShell.cs

```diff
@@ -89,6 +99,16 @@ public sealed class VsAppShell : IApplicationShell, IMainThread, IIdleTimeServic
             }
         }
 
+        private static VsAppShell GetInstance() {
```

Comment:
Perhaps we also want assert if someone calls any method on non-initialized objects.